### DISCUSSION TITLE
[4.0] Media Manager - Prompt when changing

### DISCUSSION THREAD

**Discussion by particthistle** [joomla/joomla-cms#47147] [2. Cold Features]
### Is your feature request related to a problem? Please describe.
When you rename or delete a file in the media manager, improved UX would include a prompt to remind the user that
* Renaming the file will break all existing links to the file with the current filename (Continue / Cancel)
* Deleting the file will break all existing links to the file being deleted, as well as not being undoable (Continue / Cancel)

### Describe the solution you'd like
For 4.0, add prompts when you go to rename or delete a file in the media manager, (similar to what JCE File Browser prompts with on Renaming a file, but not on deleting a file).

For 4.1, this is something also covered in the [J4 Media Manager Roadmap](https://volunteers.joomla.org/teams/new-media-manager-team/reports/411-roadmap) that would be tied into the Database Abstraction layer, which would then separate the media manager table from the filesystem (so you're then not renaming the file, you're adjusting the listing for the file).

### Additional context
Discovered while testing PR #30839

**Comment by PhilETaylor** [comment]
linking https://github.com/joomla/joomla-cms/issues/30973 incase someone is working on this, can kill the two birds with one stone.

**Comment by bembelimen** [comment]
> For 4.1, this is something also covered in the [J4 Media Manager Roadmap](https://volunteers.joomla.org/teams/new-media-manager-team/reports/411-roadmap) that would be tied into the Database Abstraction layer, which would then separate the media manager table from the filesystem (so you're then not renaming the file, you're adjusting the listing for the file).

This should be independend from the [method of the file system](https://github.com/joomla/joomla-cms/issues/31414), I think. The Roadmap from the media manager team is not the leading one currently (but still have a lot of good stuff which should be taken over).

**Comment by N6REJ** [comment]
Why not have it peruse the db and rename the files there @ the same time.

**Comment by brianteeman** [comment]
We could add a warning message in the rename dialog if it is really desired

![Image](https://github.com/user-attachments/assets/c4eb53f0-2157-4855-9cbd-064d2923dc4e)
